crates/cast/bin/cmd/run.rs

```diff
@@ -122,14 +125,14 @@ impl RunArgs {
 
         env.block.number = U256::from(tx_block_number);
 
-        let block = provider.get_block_with_txs(tx_block_number).await?;
+        let block = provider.get_block(tx_block_number.into(), true).await?;
```

Comment:
we should change this on the provider level and add

get_full_block or something

Reply:
Yeah agree, we can separate impls—let's do this in a followup as it will affect a few places